### ISSUE TITLE
TypeError on cypress open tty.getWindowSize

### ISSUE THREAD

**Issue by jennifer-shehane** [cypress-io/cypress#1815]
### Is this a Feature or Bug?

Bug

### Current behavior:

On `cypress open`, some users are getting TypeError on new update to 3.0.0

![](https://image.ibb.co/bWjSCd/screen.jpg)

```bash
TypeError: tty.getWindowSize is not a function
at Object.<anonymous> (C:\Users\Ladder Digital\AppData\Local\Cypress\Cache\3.0.0\Cypress\resources\app\packages\server\node_modules\mocha\lib\reporters\base.js:120:13)
at Object.<anonymous> (C:\Users\Ladder Digital\AppData\Local\Cypress\Cache\3.0.0\Cypress\resources\app\packages\server\node_modules\mocha\lib\reporters\base.js:489:3)
at Module._compile (module.js:569:30)
at Object.Module._extensions..js (module.js:580:10)
at Module.load (module.js:503:32)
at tryModuleLoad (module.js:466:12)
at Function.Module._load (module.js:458:3)
at Module.require (module.js:513:17)
at require (internal/module.js:11:18)
at Object.<anonymous> (C:\Users\Ladder Digital\AppData\Local\Cypress\Cache\3.0.0\Cypress\resources\app\packages\server\node_modules\mocha\lib\reporters\index.js:3:31)
at Object.<anonymous> (C:\Users\Ladder Digital\AppData\Local\Cypress\Cache\3.0.0\Cypress\resources\app\packages\server\node_modules\mocha\lib\reporters\index.js:21:3)
at Module._compile (module.js:569:30)
at Object.Module._extensions..js (module.js:580:10)
at Module.load (module.js:503:32)
at tryModuleLoad (module.js:466:12)
at Function.Module._load (module.js:458:3)
at Module.require (module.js:513:17)
at require (internal/module.js:11:18)
at Object.<anonymous> (C:\Users\Ladder Digital\AppData\Local\Cypress\Cache\3.0.0\Cypress\resources\app\packages\server\node_modules\mocha\lib\mocha.js:13:17)
at Object.<anonymous> (C:\Users\Ladder Digital\AppData\Local\Cypress\Cache\3.0.0\Cypress\resources\app\packages\server\node_modules\mocha\lib\mocha.js:504:3)
at Module._compile (module.js:569:30)
at Object.Module._extensions..js (module.js:580:10)
at Module.load (module.js:503:32)
at tryModuleLoad (module.js:466:12)
at Function.Module._load (module.js:458:3)
at Module.require (module.js:513:17)
at require (internal/module.js:11:18)
at Object.<anonymous> (C:\Users\Ladder Digital\AppData\Local\Cypress\Cache\3.0.0\Cypress\resources\app\packages\server\node_modules\mocha\index.js:3:5)
at Object.<anonymous> (C:\Users\Ladder Digital\AppData\Local\Cypress\Cache\3.0.0\Cypress\resources\app\packages\server\node_modules\mocha\index.js:5:3)
at Module._compile (module.js:569:30)
at Object.Module._extensions..js (module.js:580:10)
at Module.load (module.js:503:32)
at tryModuleLoad (module.js:466:12)
at Function.Module._load (module.js:458:3)
at Module.require (module.js:513:17)
at require (internal/module.js:11:18)
at Object.<anonymous> (C:\Users\Ladder Digital\AppData\Local\Cypress\Cache\3.0.0\Cypress\resources\app\packages\server\lib\reporter.js:11:11)
at Object.<anonymous> (C:\Users\Ladder Digital\AppData\Local\Cypress\Cache\3.0.0\Cypress\resources\app\packages\server\lib\reporter.js:402:4)
at Object.<anonymous> (C:\Users\Ladder Digital\AppData\Local\Cypress\Cache\3.0.0\Cypress\resources\app\packages\server\lib\reporter.js:404:3)
at Module._compile (module.js:569:30)
at Object.Module._extensions..js (module.js:580:10)
at Module.load (module.js:503:32)
at tryModuleLoad (module.js:466:12)
at Function.Module._load (module.js:458:3)
at Module.require (module.js:513:17)
at require (internal/module.js:11:18)
at Object.<anonymous> (C:\Users\Ladder Digital\AppData\Local\Cypress\Cache\3.0.0\Cypress\resources\app\packages\server\lib\project.js:49:14)
at Object.<anonymous> (C:\Users\Ladder Digital\AppData\Local\Cypress\Cache\3.0.0\Cypress\resources\app\packages\server\lib\project.js:654:4)
at Object.<anonymous> (C:\Users\Ladder Digital\AppData\Local\Cypress\Cache\3.0.0\Cypress\resources\app\packages\server\lib\project.js:656:3)
at Module._compile (module.js:569:30)
at Object.Module._extensions..js (module.js:580:10)
at Module.load (module.js:503:32)
at tryModuleLoad (module.js:466:12)
at Function.Module._load (module.js:458:3)
at Module.require (module.js:513:17)
at require (internal/module.js:11:18)
at Object.<anon
```

### Desired behavior:

No TypeError :)

### Steps to reproduce:

```js
.node_modules\.bin\cypress open
```

### Versions

Cypress: 3.0.0
OSes observered: Windows 10 Pro, Windows 8.1, Windows 7

**Comment by jennifer-shehane** [comment]
We recommend downgrading to 2.1.0 in the meantime, while we work on this issue. 
```bash
npm install --save-dev cypress@2.1.0
```

**Comment by gregorybleiker** [comment]
I've had a look at the mocha code at that location. It is referencing a (very) old method of node, that has been removed. Also, the referenced mocha is old (2.4.5), so I wonder if that is the problem...?

**Comment by bahmutov** [comment]
@brian-mann we really need to bump Mocha, the problem is coming from its base reporter
```javascript
var tty = require('tty')
// ...
if (isatty) {
  exports.window.width = process.stdout.getWindowSize
      ? process.stdout.getWindowSize(1)[0]
      : tty.getWindowSize()[1];
}
```
the `tty.getWindowSize` is _super old_ and it would be a shame to polyfill `tty` for this

**Comment by jennifer-shehane** [comment]
I do not believe every Windows user is having this issue. @brian-mann mentioned this maybe having something to do with `mocha-multi-reporters`

**Comment by gregorybleiker** [comment]
An other thing that is strange: If I start the cypress.exe by hand (from appdata/local/...) everything works fine... maybe the problem is with how the process is spawn...?

**Comment by brian-mann** [comment]
Nope, we actually force `isatty` to be true because we pipe data in order to fix the garbage characters in Windows and we "trick" it into thinking it's being inherit correctly (because it effectively is).

**Comment by pklejnowski** [comment]
Why this bus is closed. Is already fixed?

**Comment by jennifer-shehane** [comment]
Released in [`3.0.1`](https://docs.cypress.io/guides/references/changelog.html#3-0-1).
